src/shared/utils.ts

```diff
@@ -203,6 +215,56 @@ export class Utils {
       rl.question("", (answer: string) => {
         rl.close();
         resolve(answer);
+      })
+    });
+  }
+  
+  /*
+   * Spawn a Node child process with predefined environment variables
+   * @param command CLI Command - NO ARGS
+   * @param spawnArgs Array of arguments for CLI command
+   */
+  public static spawn(options: ServerlessSpawnOptions): Promise<void> {
+    const { serverless, commandArgs, onSigInt } = options;
+    const { command } = options;
+    // Run command from local node_modules
+    let localCommand = join(
+      serverless.config.servicePath,
+      "node_modules",
+      ".bin",
+      command
+    );
+    
+    // Append .cmd if running on windows
+    if (process.platform === "win32") {
+      localCommand += ".cmd";
+    }
+
+    const env = {
+      // Inherit environment from current process, most importantly, the PATH
+      ...process.env,
+      // Environment variables from serverless config are king
+      ...serverless.service.provider["environment"],
```

Comment:
need a test to verify that env variables will override others.  in case future refactoring accidentally change the order

Reply:
I'll probably push this off to another PR as well. Testing this spawned process is not very straight-forward